File: demo-jest-tf/tests/calculator.test.js
const calculator = require("../src/calculator");

describe("Calculator tests", () => {
  test("adding 1 + 2 should return 3", () => {
    expect(calculator.add(1, 2)).toBe(3);
  });

  test("should add negative numbers", () => {
    expect(calculator.add(-2, -3)).toBe(-5);
  });

  test("should add mixed numbers", () => {
    expect(calculator.add(-5, 10)).toBe(5);
  });

  test("subtracting 2 from 10 should return 8", () => {
    var result = calculator.subtract(10, 2);
    expect(result).toBe(8);
  });

  test("multiplying 2 and 4 should return 8", () => {
    var result = calculator.multiply(2, 4);
    expect(result).toBe(8);
  });

  test("dividing 10 by 2 should return 5", () => {
    var result = calculator.divide(10, 2);
    expect(result).toBe(5);
  });

});

File: demo_elixir_tf/test/calculator_test.exs
defmodule CalculatorTest do
  use ExUnit.Case
  doctest Calculator

  describe "add/2" do
    test "adds two positive integers" do
      assert Calculator.add(2, 3) == 5
    end

    test "adds positive and negative integers" do
      assert Calculator.add(5, -3) == 2
    end

    test "adds two negative integers" do
      assert Calculator.add(-2, -3) == -5
    end

    test "adds integers and floats" do
      assert Calculator.add(2, 3.5) == 5.5
    end
  end

  describe "subtract/2" do
    test "subtracts two positive integers" do
      assert Calculator.subtract(5, 3) == 2
    end

    test "subtracts resulting in negative number" do
      assert Calculator.subtract(3, 5) == -2
    end

    test "subtracts with negative numbers" do
      assert Calculator.subtract(-3, -5) == 2
    end

    test "subtracts integers and floats" do
      assert Calculator.subtract(5, 2.5) == 2.5
    end
  end

  describe "multiply/2" do
    test "multiplies two positive integers" do
      assert Calculator.multiply(2, 3) == 6
    end

    test "multiplies positive and negative integers" do
      assert Calculator.multiply(2, -3) == -6
    end

    test "multiplies two negative integers" do
      assert Calculator.multiply(-2, -3) == 6
    end

    test "multiplies integers and floats" do
      assert Calculator.multiply(2, 3.5) == 7.0
    end
  end

  describe "divide/2" do
    test "divides two positive integers" do
      assert Calculator.divide(6, 3) == 2.0
    end

    test "divides resulting in float" do
      assert Calculator.divide(5, 2) == 2.5
    end

    test "divides with negative numbers" do
      assert Calculator.divide(-6, 3) == -2.0
      assert Calculator.divide(6, -3) == -2.0
      assert Calculator.divide(-6, -3) == 2.0
    end

    test "raises error when dividing by zero" do
      assert_raise ArithmeticError, "division by zero", fn ->
        Calculator.divide(5, 0)
      end
    end
  end
end